src/Maui/Features/Combat/Manage.razor.cs

```diff
@@ -47,10 +49,27 @@ public void InitiativeRoll()
     {
         foreach (InitiatedCreature a in _initiatedCreatures)
         {
-            a.InititiveRoll = a.Creature.RollInitiative();
+            a.InitiativeRoll = a.Creature.RollInitiative();
         }
 
-        _initiatedCreatures = _initiatedCreatures.OrderByDescending(x => x.InititiveRoll).ToList();
+        _initiatedCreatures = _initiatedCreatures.OrderByDescending(x => x.InitiativeRoll).ToList();
+
+        if (_initiatedCreatures.Any()) _activeCreature = _initiatedCreatures.First();
```

Comment:
A linked list might be good here to just find the next one?

var initiatedCrestures = new LinkedList<Creature>();

Which should unlock behavior like

_activeCreature = initiatatedCreatures.Find(_activeCreature).Next.Value

Reply:
This will make going backwards easy too 😉

Reply:
Ahh, I've never used a LinkedList.. neat! I went ahead and replaced the list / indexing with a LinkedList. I wasn't quite sure how to handle the initiative sorting since OrderByDescending returns a List rather than a LinkedList so I'm newing a LinkedList up every time now.... thoughts for improvement?

Reply:
Yeah I see: maybe using something like AddAfter after finding the creature that has a higher or same initiative roll? Multiple ways to skin this I think

Reply:
Hmmmm.. I want to avoid adding too much logic here. What do you think of this? It's not great, but at least we wouldn't be creating garbage with new LinkedLists

public void InitiativeRoll()
    {
        foreach (InitiatedCreature creature in _initiatedCreatures)
        {
            creature.InitiativeRoll = creature.Creature.RollInitiative();
        }

        var creaturesOrderedByInitiative = _initiatedCreatures.OrderByDescending(x => x.InitiativeRoll).ToList();

        _initiatedCreatures.Clear();
        foreach (InitiatedCreature creature in creaturesOrderedByInitiative)
        {
            _initiatedCreatures.AddLast(creature);
        }

        if (_initiatedCreatures.Any()) _activeCreature = _initiatedCreatures.First();
    }

Reply:
I also feel like there's a missing piece in the middle, but I agree it's good enough for now. Shipping.